### DISCUSSION TITLE
Why append(contentsOf:) is unavailable?

### DISCUSSION THREAD

**Discussion by jeffersonsetiawan** [pointfreeco/swift-identified-collections#31] [Q&A] [answered]
```
extension IdentifiedArray {
    @available(*, unavailable, message: "use 'append(_:)' with each element to handle duplicates")
    public mutating func append<S>(contentsOf _: S)
        where Element == S.Element, S: Sequence {}
```

In my project, several people are using this helper, really curious about why making it unavailable.

Can we use this implementation?
```
public mutating func append<S>(contentsOf: S)
        where Element == S.Element, S: Sequence {
        contentsOf.forEach {
            insert($0, at: endIndex)
        }
    }
```

**Comment by smotched** [comment]
There has to be a reason right? seems it takes more lines to write the warning than the implementation

**Comment by stephencelis** [comment, marked as answer]
We originally left it off because it seemed unsafe to so easily ignore the fact that some elements may be inserted and some may not. However, it appears that `OrderedSet` implements this method, so for the sake of consistency I think we could implement it, as well. Want to submit a PR?

**Reply by jeffersonsetiawan** [reply]
PR created: https://github.com/pointfreeco/swift-identified-collections/pull/35 🙏
